xave-finance/halodao-interface
# [AMM-v2] Features/hdev 506 swap preview

## Description of Changes

This PR contains a little bit of everything about the Swap functionality
- estimate/preview the swap as you type in the amount
- main swap functionality

Other changes:
- improve token list loading
- balancer sdk integration (not used atm but keeping it here so we can continue to test/validate ideas until we finalized everything)

Note:
**Some functionalities on current Swap page has been removed temporarily for the interest of time. This version is only good for demo atm**
- swap setting: slippage, deadline, etc
- price estimation & min received
- swap confirmation modal; timer

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-506)

## How to Test

1. Use deployer wallet & connect to kovan network
2. Go to swap page
3. Make sure that token list loads
4. Select your desired token in and out, and enter an amount
5. Verify that the other token amount should update as you type in the amount either from token in or out
6. Verify that swap button becomes enabled if both amounts > 0
7. Clicking the swap button should perform the swap
8. Any errors along the way should be reported by the app
  - inline error for estimation errors
  - modal error for swap tx error

### Developer Checklist:


### Reviewers Checklist:
